### PR TITLE
Stop using astropy's catch_warnings in fitscheck

### LINKED ISSUE #10528: fitscheck should not be using astropy.tests.helper.catch_warnings
### Description

`fitscheck` uses `astropy.tests.helper.catch_warnings`:

https://github.com/astropy/astropy/blob/49708febbdaa1d1bf53f96981bf1f1d71295d33d/astropy/io/fits/scripts/fitscheck.py#L134

It should not be using a function from "test helper" module because that makes `pytest` a compulsory dependency whenever this import happens:

https://github.com/astropy/astropy/blob/49708febbdaa1d1bf53f96981bf1f1d71295d33d/astropy/io/fits/scripts/fitscheck.py#L47

### Expected behavior

Any non-test function should not be using stuff in `astropy.tests`.

### Blocker

An attempt to replace it with `warnings.catch_warnings` failed in #10504 , which @saimn explained beautifully at https://github.com/astropy/sphinx-astropy/issues/21#issuecomment-653692416 : 

*Basically Astropy's `catch_warnings` is overwriting the warnings defined in `setup.cfg` for pytest, by calling `treat_deprecations_as_exceptions` in its `__exit__` method. So the global `error` clause in pytest's config was ignored once Astropy's `catch_warnings` was called... When using `warnings.catch_warnings` this does not happen, so the global `error` clause is there, and the checksum verification in `io.fits` is issuing warning when the checksum is wrong, which turns into errors, which break the code's logic...*

### PR DESCRIPTION
**Pull request by saimn**
Fix #10528.

### CODE REVIEW

File: astropy/io/fits/scripts/fitscheck.py [hunk 1]
```diff
@@ -106,6 +105,8 @@ def handle_options(args):
 
     if OPTIONS.checksum_kind == 'none':
         OPTIONS.checksum_kind = False
+    elif OPTIONS.checksum_kind == 'standard':
```
Comment (pllim) [inline astropy/io/fits/scripts/fitscheck.py:108]: This doesn't look related.

**Comment by saimn** [comment]
> My FITS-fu is not high enough to know how this will change actual fitscheck behavior outside of CI but I'll just trust the CI. grimacing

It shouldn't change anything, just making sure that warnings are not turned into exceptions.

> My only nitpick is that the import order was alphabetical in fitscheck.py and there is no reason to change it? wink

That's the way isort is configured on the repo ;)

File: astropy/io/fits/scripts/fitscheck.py (hunk 1 above)
Comment (saimn) [reply astropy/io/fits/scripts/fitscheck.py:108]: Not really, just making sure that the option value makes sense, 'standard' evaluates as True but it has no special meaning and `True` should be used here.

**Comment by pllim** [comment]
There's a remote failure but doesn't seem related.
```
        with iers.earth_orientation_table.set(iers.IERS_B.open()):
>           with pytest.warns(AstropyWarning, match='after IERS data'):
E           NameError: name 'AstropyWarning' is not defined
../../.tox/py37-test-devdeps/lib/python3.7/site-packages/astropy/coordinates/tests/test_intermediate_transformations.py:536: NameError
```

**Comment by pllim** [comment]
I approved it but I'll leave it for someone else to merge in case more pairs of eyes are needed.
